File: src/main/java/no/nav/veilarbvedtaksstotte/config/KafkaProperties.java
package no.nav.veilarbvedtaksstotte.config;

import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

@Setter
@ConfigurationProperties(prefix = "app.kafka")
public class KafkaProperties {
    String brokersUrl;
    String endringPaAvsluttOppfolgingTopic;
    String endringPaOppfolgingsBrukerTopic;
    String vedtakSendtTopic;
    String vedtakStatusEndringTopic;
    String arenaVedtakTopic;
    String siste14aVedtakTopic;
    String vedtakFattetDvhTopic;

    public String getBrokersUrl() {
        return this.brokersUrl;
    }

    public String getEndringPaAvsluttOppfolgingTopic() {
        return this.endringPaAvsluttOppfolgingTopic;
    }

    public String getEndringPaOppfolgingsBrukerTopic() {
        return this.endringPaOppfolgingsBrukerTopic;
    }

    public String getVedtakSendtTopic() {
        return this.vedtakSendtTopic;
    }

    public String getVedtakStatusEndringTopic() {
        return this.vedtakStatusEndringTopic;
    }

    public String getArenaVedtakTopic() {
        return this.arenaVedtakTopic;
    }

    public String getSiste14aVedtakTopic() {
        return this.siste14aVedtakTopic;
    }

    public String getVedtakFattetDvhTopic() {
        return this.vedtakFattetDvhTopic;
    }
}

File: src/main/java/no/nav/veilarbvedtaksstotte/config/KafkaProperties.kt
package no.nav.veilarbvedtaksstotte.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "app.kafka")
data class KafkaProperties(
    val brokersUrl: String,
    val arenaVedtakTopic: String,
    val endringPaOppfolgingsBrukerTopic: String,
    val siste14aVedtakTopic: String,
    val sisteOppfolgingsperiodeTopic: String,
    val vedtakFattetDvhTopic: String,
    val vedtakSendtTopic: String,
    val vedtakStatusEndringTopic: String,
    val pdlAktorV2Topic: String,
    val gjeldende14aVedtakTopic: String
)
